### DISCUSSION TITLE
How to populate User Fields from OAuth2 provider?

### DISCUSSION THREAD

**Discussion by PredragPeranovic** [fief-dev/fief#268] [Q&A] [answered]
Hi,

Google provider has the following fields inside UserInfo  `displayName`, `photoUrl`, `dateOfBirth` etc. 

How can I read and map values from those fields to Fief User Fields on the first login using the Google provider?

All the best, and thank you.

**Comment by frankie567** [comment, marked as answer]
This is not something we implemented right now.

You can do this on your side by using the Fief's Admin API: 

* A first call to get a fresh Google access token for the user (https://docs.fief.dev/going-further/oauth-provider-token/)
* A second call to update user fields (https://example.fief.dev/admin/api/docs#/Users/users_update_users__id__patch)
